projects/netgrif-components-core/src/lib/data-fields/case-ref-field/abstract-case-ref-field.component.ts

```diff
@@ -0,0 +1,250 @@
+import {AfterViewInit, Component, Input} from '@angular/core';
+import {
+    PetriflowArc,
+    PetriflowCanvasConfigurationService,
+    PetriflowCanvasFactoryService,
+    PetriflowCanvasService,
+    PetriflowInhibitorArc,
+    PetriflowPlace,
+    PetriflowPlaceTransitionArc, PetriflowReadArc,
+    PetriflowResetArc,
+    PetriflowTransition, PetriflowTransitionPlaceArc
+} from '@netgrif/petriflow.svg';
+import {
+    Arc,
+    InhibitorArc, Place,
+    ReadArc,
+    RegularPlaceTransitionArc,
+    RegularTransitionPlaceArc,
+    ResetArc
+} from '@netgrif/petri.svg';
+import {CaseRefField} from './models/case-ref-field';
+import {PetriNetResourceService} from '../../resources/engine-endpoint/petri-net-resource.service';
+import {CaseResourceService} from '../../resources/engine-endpoint/case-resource.service';
+import {TransitionImport} from '../../resources/interface/transition-import';
+import {PlaceImport} from '../../resources/interface/place-import';
+import {ArcImport} from '../../resources/interface/arc-import';
+import {ArcType} from '../../resources/interface/arc-type';
+import {PetriNetImport} from '../../resources/interface/petri-net-import';
+import {LoggerService} from '../../logger/services/logger.service';
+import {SnackBarService} from '../../snack-bar/services/snack-bar.service';
+import {TranslateService} from '@ngx-translate/core';
+
+@Component({
+    selector: 'ncc-abstract-case-ref-field',
+    template: ''
+})
+export abstract class AbstractCaseRefFieldComponent implements AfterViewInit {
+
+    @Input() public dataField: CaseRefField;
+
+    constructor(protected _petriflowCanvasService: PetriflowCanvasService, protected _petriflowFactoryService: PetriflowCanvasFactoryService,
+                protected _petriflowConfigService: PetriflowCanvasConfigurationService, protected _caseResourceService: CaseResourceService,
+                protected _petriNetResourceService: PetriNetResourceService, protected _log: LoggerService, protected _snackBar: SnackBarService,
+                protected _translate: TranslateService) {
+    }
+
+    ngAfterViewInit(): void {
+        this._petriNetResourceService.getNetByCaseId(this.dataField.value).subscribe(net => {
```

Comment:
Must the net always be get like this? Can't the process service cache be utilised in some way, so that multiple calls to the backend are not made?

moreover, the `this.dataField.value` is ALWAYS an `Array` for caseRef fields and the wording of the called method implies a `string` argument

the value should be read from the `FormControl` so that it gets updated with processed `changedFields` from other tasks

Reply:
done